app/assets/javascripts/host_edit.js

```diff
@@ -363,11 +366,14 @@ function architecture_selected(element){
 }
 
 function os_selected(element){
-  var attrs = attribute_hash(['operatingsystem_id', 'organization_id', 'location_id']);
   var url = $(element).attr('data-url');
+  var type = $(element).attr('data-type');
+  var obj = (type == "hosts" ? "host" : "hostgroup");
+  var attrs = {};
+  attrs[obj] = attribute_hash(['operatingsystem_id', 'organization_id', 'location_id']);
```

Comment:
May be out of scope but would it be possible for you  to have a global var or some mechanism by which you can make attribute_hash accept additional set attributes to send when os gets selected.  

Basically Katello needs these also to get sent when os gets selected
```
  var attrs = attribute_hash(['operatingsystem_id', 'organization_id', 'location_id', 'content_view_id',
                              'lifecycle_environment_id', 'content_source_id', 'architecture_id', 'hostgroup_id',
                              'medium_id', 'kickstart_repository_id']);
```

In fact we have to hack and over ride the complete method just for that additional set of attributes. Look at -> https://github.com/Katello/katello/blob/master/app/assets/javascripts/katello/hosts/host_and_hostgroup_edit.js#L252-L274

If there was ether a "os_attributes" or some global param we could very easily add the 3 params needed and not duplicate this code it would be great . As a bonus it will also fix -> http://projects.theforeman.org/issues/14699